flagsmith/models.py

```diff
@@ -39,7 +55,7 @@ def from_feature_state_model(
         )
 
     @classmethod
-    def from_api_flag(cls, flag_data: dict) -> "Flag":
+    def from_api_flag(cls, flag_data: typing.Mapping[str, typing.Any]) -> Flag:
```

Comment:
I tried adding a JsonType here but got a bunch of errors. If we want to do something like `flag_data["key"]["nested-key"]` then we will have to narrow the type up to 2 levels of nesting. Narrow the type to dict for `flag_data["key"]` and then narrow it further to dict for `flag_data["key"]["nested-key"]` as JsonType could be any of `int | None | str | JsonType ...`

Hovering over JsonType reveals a sizable recursive type hint. Would it be clearer for users if we defined types reflecting the structure of responses from various endpoints instead of relying on a generic JsonType?

Reply:
This is a great point Tushar. Unfortunately, our API is not quite ready yet. We'll work on out v2 API that will export contracts suitable for code generation soo!. For now, we have to stick with generic types.